plugin/src/main/java/org/astropeci/omw/listeners/ItemDeployHandler.java

```diff
@@ -103,69 +105,46 @@ private boolean deployStructure(GameTeam team, Location target, PlayerInteractEv
         int height;
         int length;
 
-        switch (material) {
-            case CREEPER_SPAWN_EGG:
-                structureName = "tomahawk";
-
-                offsetX = 0;
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 2;
-                height = 2;
-                length = 13;
-                break;
-            case GUARDIAN_SPAWN_EGG:
-                structureName = "guardian";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 3;
-                height = 3;
-                length = 8;
-                break;
-            case GHAST_SPAWN_EGG:
-                structureName = "juggernaut";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 3;
-                height = 3;
-                length = 11;
-                break;
-            case WITCH_SPAWN_EGG:
-                structureName = "shieldbuster";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 4;
-
-                width = 3;
-                height = 3;
-                length = 15;
-                break;
-            case OCELOT_SPAWN_EGG:
-                structureName = "lightning";
-
-                offsetX = 1;
-                offsetY = 4;
-                offsetZ = 5;
-
-                width = 3;
-                height = 2;
-                length = 9;
-                break;
-            default:
-                return false;
+        ConfigurationSection missilesC = plugin.getConfig()
+                .getConfigurationSection("missiles");
+
+        if(missilesC == null)
+            return false;
+
+        ConfigurationSection missileConfig = missilesC.getConfigurationSection(material.name());
+
+        if(missileConfig == null)
+            return false;
+        else{
+            structureName = missileConfig.getString("structureName");
+
+            offsetX = missileConfig.getInt("offsetX");
+            offsetY = missileConfig.getInt("offsetY");
+            offsetZ = missileConfig.getInt("offsetZ");
+
+            width = missileConfig.getInt("width");
+            height = missileConfig.getInt("height");
+            length = missileConfig.getInt("length");
```

Comment:
Might be nice to cache this as opposed to looking it up every time a missile is spawned. There should also be logging in the event that a missile model does not exist. I would also suggest that `missilesC` could be renamed to something clearer.

Reply:
Only the part up until `if(misiilesC == null)` can be cached, because the next call depends on `material`which is a function parameter. Also logging is probably not a good idea, since this function gets called whenever a player right-clicks on a block, even if they're holding a bow or nothing at all.

Reply:
Oh yes your right on both points, sorry hadn't read it through properly.